srcopsmetrics/iterator.py

```diff
@@ -84,16 +86,13 @@ def run(self):
         try:
             entities = self.entity.analyse()
             length = entities.totalCount if isinstance(entities, PaginatedList) else len(entities)
-            for idx, entity in enumerate(tqdm(entities, total=length), 1):
-                self.knowledge_updated = True
-
-                remaining = self.github.get_rate_limit().core.remaining
 
-                if remaining <= API_RATE_MINIMAL_REMAINING:
-                    self.wait_until_api_reset()
+            progressbar = tqdm(entities, total=length)
+            for idx, entity in enumerate(progressbar, 1):
+                self.knowledge_updated = True
 
-                if idx % 5 == 0:
-                    _LOGGER.info("[API requests remaining: %d]" % remaining)
+                self.handler.check_and_wait_for_api()
+                progressbar.set_postfix(ordered_dict={"RATE remaining": self.handler.remaining})
```

Comment:
@harshad16 here it is constantly updated in progressbar to display the current api limit

Reply:
ack, :+1: